parlai/core/torch_agent.py

```diff
@@ -433,6 +437,13 @@ def add_cmdline_args(cls, argparser):
         agent.add_argument(
             '--fp16', type='bool', default=False, help='Use fp16 computations.'
         )
+        agent.add_argument(
+            '--fp16-impl',
+            type=str,
+            default='apex',
+            choices=['apex', 'mem_efficient'],
```

Comment:
Under what conditions would we still want to use apex? Maybe for the optimizers that this won't work with?

Reply:
Yeah. So I should write more about the tradeoffs inside the class docstring, but basically (1) memory efficinet is not compatible with everything and (2) you want to use apex if memory constraints are not an issue, as there are stability issues with the memory efficient optimizer